app/bod/management/commands/bod_migrate.py

```diff
@@ -0,0 +1,55 @@
+from bod.models import ContactOrganisation
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from provider.models import Provider
+
+
+class Command(BaseCommand):
+    help = "Migrates data from a BOD database"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--clear",
+            action="store_true",
+            help="Delete existing objects before importing",
+        )
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Dry run, abort transaction in the end",
+        )
+        parser.add_argument(
+            "--verbose",
+            action="store_true",
+            help="Show more output",
+        )
+
+    def handle(self, *args, **options):
+        with transaction.atomic():
+            if options["clear"]:
+                deleted, _ = Provider.objects.all().delete()
+                self.stdout.write(self.style.SUCCESS(f"{deleted} providers deleted"))
+
+            added = 0
+            for org in ContactOrganisation.objects.all():
+                Provider.objects.create(
```

Comment:
I'd probably avoid setting the id of the Provider explicitly. If you want to use the `pk_contactorganisation_id` as unique identifier to check if the provider already exists, I'd add it explicitly as a "bod legacy" field. For the case of provider we could also define the `abkuerzung_de` as unique identifier (and leave the `id` field to auto-magic from django). You could also use `Provider.objects.get_or_create()` then, this would only create new objects if they don't exist yet, the other fields could just be updated if the object already exists.

Reply:
Good idea! I added a legacy field to track the relationship to the old object so that it's possible to allow updates of already migrated objects.